src/GraphQL.DataLoader/DataLoaderPair.cs

```diff
@@ -60,15 +73,22 @@ public void SetResult(T value)
         /// <param name="cancellationToken">Optional <seealso cref="CancellationToken"/> to pass to fetch delegate</param>
         public async Task<T> GetResultAsync(CancellationToken cancellationToken = default)
         {
-            if (!IsResultSet)
-                await Loader.DispatchAsync(cancellationToken).ConfigureAwait(false);
+            var loader = _loader;
+            if (loader != null)
```

Comment:
Why not `IsResultSet` here and below?

Reply:
Need to eliminate a race condition where another thread may clear _loader.  IsResultSet becomes redundant.